sdk/eventhub/event-hubs/src/connectionContext.ts

```diff
@@ -59,6 +97,26 @@ export interface ConnectionContextOptions extends EventHubClientOptions {
   managementSessionAudience?: string;
 }
 
+/**
+ * Helper type to get the names of all the functions on an object.
+ */
+type FunctionPropertyNames<T> = { [K in keyof T]: T[K] extends Function ? K : never }[keyof T];
```

Comment:
So...beautiful.....

Reply:
I wish I could take credit but that's right out of the TypeScript handbook: https://www.typescriptlang.org/docs/handbook/advanced-types.html#distributive-conditional-types